go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/atlassian/gostatsd
 
-go 1.19
+go 1.21.4
```

Comment:
I have not upgraded the go version. This has been adjusted to match the go version in other files (e.g. Docker, README, and CI config).